udev/10-azure-unmanaged-sriov.rules

```diff
@@ -0,0 +1,6 @@
+# Azure VMs with accelerated networking may have MANA, mlx4, or mlx5 SR-IOV devices which are transparently bonded to a synthetic
+# hv_netvsc device.  Mark devices with the IFF_SLAVE bit set as unmanaged devices:
+#   AZURE_UNMANAGED_SRIOV=1 for 10-azure-unmanaged-sriov.network
+#   ID_NET_MANAGED_BY=unmanaged for systemd-networkd >= 255
+#   NM_UNMANAGED=1 for NetworkManager
+SUBSYSTEM=="net", ACTION!="remove", DRIVERS=="mana|mlx4_core|mlx5_core", ATTR{flags}=="0x?[89ABCDEF]??", ENV{AZURE_UNMANAGED_SRIOV}="1", ENV{ID_NET_MANAGED_BY}="unmanaged", ENV{NM_UNMANAGED}="1"
```

Comment:
so https://github.com/Azure/WALinuxAgent/pull/1622 has `DRIVERS=="hv_pci"`. I'm guessing that `DRIVERS=="mana|mlx4_core|mlx5_core"` is a more precise version of that?

Reply:
`ATTR{flags}=="0x?[89ABCDEF]??"` checks the slave-bit on VF devices that are bonded with the synthetic `hv_netvsc` link.

`DRIVERS=="hv_pci"` in WALinuxAgent matches any device with `hv_pci` as parent but the rule then can't access the flags field. This seems to only work when the drivers are listed explicitly (see this comment: https://github.com/Azure/azure-vm-utils/pull/41#discussion_r1784545666).

Reply:
> ATTR{flags}=="0x?[89ABCDEF]??" checks the slave-bit on VF devices that are bonded with the synthetic hv_netvsc link.

maybe this would be a good comment to add to the rule